### ISSUE TITLE
about baseline-on-migrate is invalid

### ISSUE THREAD

**Issue by jonahzheng** [flyway/flyway#2465]
###### Which version and edition of Flyway are you using?
5.2.4 or 6.0.0-beta2

###### If this is not the latest version, can you reproduce the issue with the latest one as well?
(Many bugs are fixed in newer releases and upgrading will often resolve the issue)
yes

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
maven plugin + spring boot 6.0.0-beta2

###### Which database are you using (type & version)?
mysql 5.7.x

###### Which operating system are you using?
mac

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, the SQL statement that failed (if relevant) and the command you ran.)
==setting==
application.yml
spring:
  flyway:
    baseline-on-migrate: true

==issue==
/Library/Java/JavaVirtualMachines/jdk-12.0.1.jdk/Contents/Home/bin/java -Dmaven.multiModuleProjectDirectory=/Users/lvzx/java/mit "-Dmaven.home=/Applications/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3" "-Dclassworlds.conf=/Applications/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3/bin/m2.conf" -Dfile.encoding=UTF-8 -classpath "/Applications/IntelliJ IDEA.app/Contents/plugins/maven/lib/maven3/boot/plexus-classworlds-2.5.2.jar" org.codehaus.classworlds.Launcher -Didea.version2019.1.2 clean install
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building mit 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ microservices_Instance_template ---
[INFO] 
[INFO] --- yaml-properties-maven-plugin:1.1.3:read-project-properties (default) @ microservices_Instance_template ---
[INFO] 
[INFO] --- flyway-maven-plugin:6.0.0-beta2:migrate (default) @ microservices_Instance_template ---
[INFO] Flyway Community Edition 6.0.0-beta2 by Boxfuse
[INFO] Database: jdbc:mysql://114.116.23.236:3306/guns_test (MySQL 5.7)
[INFO] Successfully validated 1 migration (execution time 00:00.273s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.061 s
[INFO] Finished at: 2019-08-16T17:15:02+08:00
[INFO] Final Memory: 29M/107M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:6.0.0-beta2:migrate (default) on project microservices_Instance_template: org.flywaydb.core.api.FlywayException: Found non-empty schema(s) `guns_test` without schema history table! Use baseline() or set baselineOnMigrate to true to initialize the schema history table. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by MikielAgutu** [comment]
If you want to use Flyway with maven you need to specify the configuration in the `pom.xml` file. The Spring Boot `.yml` file will not be taken into account. [See the docs](https://flywaydb.org/documentation/maven/migrate).

**Comment by jonahzheng** [comment]
@MikielAgutu https://stackoverflow.com/questions/55196889/baseline-on-migrate-with-flyway-and-spring-boot

Which version of Spring Boot are you using ? – Kedar Joshi Mar 16 at 13:03
Oh you are absolutely right! I was using 2.1.3 (with Java 11). I had heard that most of those compatibility issues had been resolved but I guess not. I downed it to 1.5.19 (and Java 8) and it's working fine now. Know of any way to get around that though? Thanks so much for setting me straight! – joellis13 Mar 17 at 15:04
